notebook/static/edit/js/editor.js

```diff
@@ -54,6 +56,14 @@ function(
             );
             that._set_codemirror_options(cmopts);
             that.events.trigger('config_changed.Editor', {config: that.config});
+            if (cfg.file_extension_modes) {
+                // check for file extension in user preferences
+                var modename = cfg.file_extension_modes[that._get_file_extension()];
+                if (modename) {
+                    var modeinfo = CodeMirror.findModeByName(modename);
+                    that.set_codemirror_mode(modeinfo);
```

Comment:
Does this need to handle CodeMirror not finding the mode? That might be possible if you switch between two versions of the notebook, for instance.